api/v1beta1/infraenv_types.go

```diff
@@ -103,6 +103,11 @@ type InfraEnvSpec struct {
 	// Applicable for both iPXE, and ISO streaming from Image Service.
 	// +optional
 	KernelArguments []KernelArgument `json:"kernelArguments,omitempty"`
+
+	// IronicAgentImageOverride is the pull spec for the ironic agent image to use when discovering hosts
+	// Only applies when provisioning using a BareMetalHost
+	// +optional
+	IronicAgentImageOverride string `json:"ironicAgentImageOverride,omitempty"`
```

Comment:
Do we want this directly in the spec, or would an annotation be more appropriate?

Reply:
I made the case elsewhere for putting it in the Spec, and I think @avishayt agreed.

Reply:
Re-reading, maybe I spoke too soon. :) My basic assertion is that putting it in the spec:
* lets us clearly document what the feature is and how/when to use it
* helps users discover that the feature exists
* helps users notice when the feature is in use (people look at the spec on existing resources to see input that affects an API's behavior; it's easy to miss input that's provided elsewhere, such as an annotation)
* makes it clear how the field will evolve in the future. If it get deprecated, replaced, or changes in other ways, having it in the Spec gives us a clear process for managing that change.
* lets us use k8s validation features
* provides an opportunity to put corresponding info in the Status